fili-core/src/main/java/com/yahoo/bard/webservice/druid/model/query/GroupByQuery.java

```diff
@@ -206,7 +207,7 @@ public GroupByQuery withIntervals(Collection<Interval> intervals) {
 
     @Override
     public GroupByQuery withAllIntervals(Collection<Interval> intervals) {
-        Optional<DruidFactQuery<?>> innerQuery = (Optional<DruidFactQuery<?>>) this.dataSource.getQuery();
+        Optional<DruidFactQuery<?>> innerQuery = this.dataSource.getQuery().map(DruidFactQuery.class::cast);
```

Comment:
@michael-mclawhorn For example, this will properly cast at runtime.  Generic casting of Optional is only a compiler time workaround due to type erasure.